inc/class-tmsc-custom-landing-page-types.php

```diff
@@ -0,0 +1,816 @@
+				'name' =>  $this->current_taxonomy,
+				'orderby' =>  'name',
+				'hierarchical' =>  true,
+				'depth' =>  1,
+				'show_count' =>  false,
+				'hide_empty' =>  false,
+				'hide_if_empty' => true,
+				'selected' => $selected,
+			) );
+		}
+	}
+
+	/**
+	 * This is used to display the children of a post when the edit.php is passed with a filter action.
+	 *
+	 */
+	public function parse_query( $query ) {
+		$this->set_current_landing_page_meta();
+		global $pagenow, $typenow;
+		if (
+			'edit.php' === $pagenow &&
+			in_array( $typenow, array_keys( $this->linked_types ), true )
+		) {
+			// These allows a post type that has been assigned to multiple taxonomies to return the proper results on the orginal menu link.
+			if ( empty( $this->current_taxonomy ) ) {
+				$query->set( 'post_parent', 0 );
+				$query->set( 'meta_query', array(
+					'relation' => 'OR',
+					array(
+						'key' => 'linked_taxonomy',
+						'value' => $this->taxonomies,
+						'compare' => 'NOT IN',
+					),
+					array(
+						'key' => 'linked_taxonomy',
+						'compare' => 'NOT EXISTS',
+					),
+				) );
+			} elseif ( ! empty( $this->current_taxonomy ) && in_array( $this->current_taxonomy, $this->taxonomies, true ) ) {
+				if ( ! empty( $query->query['post_status'] ) ) {
+					$query->set( 'meta_query', array(
+						array(
+							'key' => 'linked_taxonomy',
+							'value' => $this->current_taxonomy,
+						),
+					) );
+					unset( $query->query_vars[ $this->current_taxonomy ] );
+					unset( $query->query_vars[ 'taxonomy' ] );
+					unset( $query->query[ $this->current_taxonomy ] );
+					unset( $query->query[ 'taxonomy' ] );
+					$query->set( 'orderby', 'menu_order title' );
+					$query->tax_query = null;
+				} elseif ( ! empty( $this->current_term ) ) {
+					$parent = tmsc_get_linked_post( $this->current_term->term_id );
+					$query->set( 'post_parent', $parent->ID );
+					unset( $query->query_vars[ $this->current_taxonomy ] );
+					unset( $query->query[ $this->current_taxonomy ] );
+					$query->tax_query = null;
+				} else {
+					// Are we on the 'All' status page?
+					if ( empty( $_GET['landing_type'] ) ) {
+						$query->set( 'order', 'asc' );
+						$query->set( 'orderby', 'menu_order title' );
+						$query->set( 'fields', 'id=>parent' );
+					} elseif ( 'single' === $this->current_landing_type ) {
+						$query->set( 'post_parent__not_in', array( 0 ) );
+					} else {
+						$query->set( 'post_parent', 0 );
+					}
+					$query->set( 'meta_query', array(
+						array(
+							'key' => 'linked_taxonomy',
+							'value' => $this->current_taxonomy,
+						),
+					) );
+
+				}
+			}
+		}
+
+		return $query;
+	}
+
+	/**
+	 * Make sure we are redirecteds with the correct landing page query args
+	 */
+	public function set_redirect_query_args( $location, $post_id ) {
+		$post = get_post( $post_id );
+		if ( $this->is_custom_landing_page_post( $post ) ) {
+			$meta = get_post_meta( $post_id );
```

Comment:
`get_post_meta` is an expensive call and if you are not using it then I would remove it.